### DISCUSSION TITLE
Way to execute an action when all input buttons have been released?

### DISCUSSION THREAD

**Discussion by sabudum** [sezanzeb/input-remapper#908] [Q&A] [answered]
I'm trying to create an Alt+Tab functionality for my controller.
When I press L1+L2+Select, it executes an Alt+Tab command, however, this makes the little alt+tab window disappear immediately after releasing Select, which makes it hard to choose the correct window.

What I want to do is, when I press L1+L2+Select, an Alt+Tab command is launched, but if I release Select, Alt should remain pressed, until I release L2 or L1.

Here's the thing, I can't just remap Alt to L1+L2 and Tab to Select. Because I need Select on it's own to be mapped to Spacebar, and I need L1+L2 as a modifier for other keys, for instance L1+L2+X is mapped to KEY_K.

Is there a way to achieve this kind of behavior?

**Comment by sabudum** [comment, marked as answer]
Solved it by creating the remaps as follow:
L1+L2                   mapped to macro                 hold().key_up(KEY_LEFTALT)
L1+L2+Select       mapped to macro                 key_down(KEY_LEFTALT).key(KEY_TAB)

This makes the combo L1+L2 always release the Alt key when released, which is not a problem if Alt is not being pressed.
And the combo L1+L2+Select always holds Alt down, which will be released when L1+L2 is also released.

**Comment by zany130** [comment]
What I want to know is how are you even using L2 as a mapping? for me, I can't use any analog input (sticks ,dpad, triggers) as that disables the input for the original controller

so if I make a  select + L2 binding I can't use L2 anymore by itself in games see https://github.com/sezanzeb/input-remapper/issues/893

@sabudum
